src/hash_map.c

```diff
@@ -230,6 +230,20 @@ static rcutils_ret_t hash_map_check_and_grow_map(rcutils_hash_map_t * hash_map)
   return ret;
 }
 
+// Modified from http://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
+static size_t next_power_of_two(size_t v)
+{
+  size_t shf;
+  v--;
+  for (size_t i = 0; i <= sizeof(size_t); ++i) {
+    shf = (1 << i);
+    v |= v >> shf;
+  }
+  v++;
+
+  return v > 1 ? v : 1;
+}
+
```

Comment:
```suggestion
// Modified from http://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
static size_t next_power_of_two(size_t v)
{
  size_t shf = 0;
  v--;
  for (size_t i = 0; shf < sizeof(size_t) * 4; ++i) {
    shf = (1 << i);
    v |= v >> shf;
  }
  v++;
  return v > 1 ? v : 1;
}

```

My bad for initially introducing buggy code...

The original version (and the edit) causes an overflow/runtime error.

![image](https://user-images.githubusercontent.com/20265670/230517295-80b3ac91-def1-454c-b472-e876a3e2281c.png)


[Try it on godbolt](https://godbolt.org/#z:OYLghAFBqd5QCxAFwE4FN0BoCWIIDGAtgIYDW6AgqsAM4gDkAtACIDCAspQNICiA%2BgCEAqgEkAMi34AVAJoAFXgFIAzCwBK6ADYB1HMgQt0AI1EA7AGYB7BlgJWiABxxb0qcSTPAAriWDpRABMQAiUAJkFwwTtvWmQHAHlvZEdkgDEXdDMSInQUBHR%2BdAAPdAJkkmNXLAtM%2Blh0QP0rVCCQADYazLaARgAWLtds3JBSHDMAOgJHRyxQBlscWnMCLW9AxsYABkXaDhJxgGUrb1QCPIYe3YTHLO3bLU9gNoWASiwIRubW4J7BgN%2BAFZ/sM8pxyOhxEtkLQJshisg5oxHKgrAArMrICC0BAkW60V5KLaUImUYgQ/hEcY4IjeIj8DAAR28OAwgQgADVeOpDqIEgA5ADUKgmgMJxNJJECgSKpXKyEqrmxuPxzJIGFJgq1gpxePQtAA9ARWat0FMZprtbr8QaMAQFV5XObHJatWNJtNHOKScSFTR0Mh%2BFpxmQgzhjKh1Th9cq9bQ1RridrBRYiMhAt7dis1htggwdrg9gczMdTud7oWbnd8w8nm1yRR3p8mvEfiAwv9ep1akMcnkMLQTmd9QaClotFY4QikQxSQAJbQTrCCgDuLS0gTAYFJpKYe6YpOkBUFvDMgVJWbMq3Wmxrhf2RyH5cu11uZgrjy89aiUSbX1bbQqJ2vwAJwgn2IA4JOnozpe165hWSwPiWT4XFclZvh%2BdZ5gwf4ti0bQDD2ALtls4EjFBEwILBhbZjeeYFkhxalsOjDoUsVbvnen7PDhrzvEs4IUPIqIYvaFZoiAfxmOgK4ZK4bRhD07xaIwgK2FxBZWIw0gYOggrhCo6HIPc/FYGQpGin0ADsIF9ICYRbICNl2c5WCqQwfQaRW2kLFg9BkSZNZNjAUChRASD2BsvD4aguD4MRoKjMWzo1FYZjIIcBAkK4UlEelyDCLQ6DyMUEkdO5TwhD%2BET%2BdomJQe%2BnxnmwVhrEQ75/FkgRQjJ/J0sYbggAAHFgjhWLQ%2BiNa17WdWNE1TelvXoP1RCDagI11a49qNYcfrIDNdJzcV23ILt%2B3Lat62bXE6oHW1R1Sf5F3jCtA1DcN7yDmWFyGeM8H6aobCpsgRotGaCCqMoEoSmEKj/TmgNhGElGQ8jO7EuMyCCu6ECvAZ1mREm2ogyAIAoljEDhGE8n6fYGVZDCIBKICbBmEohMcyw1PLv4gaBCQCp44SKhEyS1nc8SuHuWp3l3tp7qpUFOxNpFVjRbF8UQIlEFCZC0KwvCiLWBlWU5Xk/RpRlRUlWVd6SX8PEhKQFBbQ16X4N1h0dSA1lYN1l3vRtnTjZNZ3pd775%2B6Hi1mIHa1DZ0J3uyW%2B2R77bs7ele13fH11J2nD0%2B37t2oMgeeJ19qHIqJmKxvi3qkg2hRUmYNJ0gy6DMqyjSctyvICsKoqN8SUoyiUZQVFU6AQAYhQT/KiroK6gqUZ6K%2BK56I%2BUH6fNhmYobBhGUYxnPsqTwq08ryDGYXipstYJptjaXrUJxIb07K9LEUhOr6Axd8LWOsRgDlQoaMcE4pzGwKmbXKlsTaFWKqVcqHYnahAiFEfy1dZzEgXOOKwy41yoA3FuDGlB9wHmJEefSp5zxS3vgwdSj8fIgFAT9cBi5JxGywF/VWv8NaALwNrTISU14zCtplbKcD8rWyQXbAskkgJoJqtEZOWcmpeyLp1ICAdXpXUTvNMO00tFSSAjHcOcc9FBwqmoixOcy7px6EBWx51c5WITsHZ6d1HHOJen1ax7Qq4/UYH9K8iMDIqGBnEDAOQoY7jhgjG8ES2DRPGMAOJMNKBxGCCAVJXhBR7wFkLfGHMxbJmyWTHAFhonoByKvapVNkZsOHBw/BUDqYizKdqWIaTBSiGxqoFgOp0yVJqTkYw3gLD8H0G4QWLQgYEFxKgOJosV4YGQKcMwwycl5OABAfpEAqkEmXAc/iqgxZczvjLRhcstKQWguI3hHw1YCNbEAkREFUYSNgRbGRiDbblQGMojBtUXEe2aoERx/szwVw2tHBaFioXmMarCjOYLU7eJMX8dF9jy7uOuiXQus0nqlzxf4jxvsgmsVnAkiwZ50AWEFGkBICR%2BBzniSoDYtQZJMpZWyshoSAbJN2Rkn0WSRm5LQL0wpgsSB43OQKuG3UqlXI8kwp%2BWBtJfN4e8AoUohp43MiAPow0Jh9HNRay1lrrleSwEQEAgIyIat8rYAKPDTIfHCuFNWThMhxSEfYX1rgNrAB6FsICPZkBuHoBAYwFZjDjHVAATzYrYQNuQMoJDMFoFNd4NikEdBcAsTQ7RnQAG76grAvZIaFFgMw8gWI%2BkZUBJtahsCsUr7UvneBYR4dAOTRhXJxVN3ypG/IkTbZB9sQBKKqugyItUj71CgElEGWAK0bQYNZJxuF3hWEcBY%2BgzBsmDPnVERygomAJBUJe3t8RXX1XUZ7FqWLoU9XxUNP4yKI6vu/ZY8l11sWPrsUSx6QHTquLLqi7FoGfYwbcQBz9VLnzGT0lJBh6qWEMGKMNdoTB2h9AKQQAggow0TBvRANgKjly6UwAZOGfxBStSDW4ejKgVCvFsDqrAeqNgbUNRZQEIpnLtC2CBECYQQJbGssNLYTigWMFtfavoWwrLDRUCp5y6nNOdGdYwV1IBAoergLAJAJwUjJCATAsdeUJ1yPKkBNAmAcC/CwCuSMjhU0Yduc/RgSQLPYysIynDeGCNEZI2RijzHnDBrYxxrjpldW1L4/gBhSn2ztAmD0Ho6mwjtGy7l/LHY9N%2BTdU8kziAQDVreUI9UdB5CeG0IcBAVgVwasDTFwWjUGsyS0M11rXF0Idb9SAUNPQOzDeDf5/rbWR0L2oHQVh%2BowGjk4VA/2co9pNC4rYBBPzbMIMnfI2wDtKpeGqiC6IS74AQFXWmddQ0t07rMvuw9jAmAnrUGeiIF6r03qYHemwWDgkvmettnrTWWuzdB05i2bmPNeeuZh%2BWjBeBymSC0QUIX8OEeAMR0jPQJhhEFJR6jgpaNIyMpx91wU0u2Htex8j7Hmcs5Z8wlHpXDM05Vu8QThPAQC8F0LwXnQPIqB85q/T3Pd1GtsmahyTkXL2UcoCUXjAwj06klsJ1WGEu0%2BuRr9ndy/LcY3ZND2fQgA%3D%3D%3D)

Reply:
To further convince myself, I tested it with input (That is, 2 ^ 62 + 1):

```
0b0100000000000000000000000000000000000000000000000000000000000001
```

To see if the for loops are all necessary:
```
v: 0100000000000000000000000000000000000000000000000000000000000000
v: 0110000000000000000000000000000000000000000000000000000000000000
v: 0111100000000000000000000000000000000000000000000000000000000000
v: 0111111110000000000000000000000000000000000000000000000000000000
v: 0111111111111111100000000000000000000000000000000000000000000000
v: 0111111111111111111111111111111110000000000000000000000000000000
v: 0111111111111111111111111111111111111111111111111111111111111111
9223372036854775808
```

Checks out.
https://godbolt.org/z/f8xaz5z98

Reply:
Nice, thanks for the update.  I've merged it now.

I was thinking of adding a test for the very large output, but `next_power_of_two` is currently static and I don't want to call `rcutils_hash_map_init` with a very large number (that will attempt to allocate a ridiculous amount of memory).  So I just took your suggestion as-is, let me know what you think.